staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -401,6 +401,9 @@ func (config *DirectClientConfig) ConfirmUsable() error {
 	} else {
 		contextName = config.config.CurrentContext
 	}
+	if config.overrides != nil && len(config.overrides.CurrentContext) != 0 {
```

Comment:
you can reuse the [getContextName](https://github.com/kubernetes/kubernetes/blob/d5fdf3135e7c99e5f81e67986ae930f6a2ffb047/staging/src/k8s.io/client-go/tools/clientcmd/client_config.go#L426-L437) method